src/heartbeat/listener.js

```diff
@@ -24,14 +26,34 @@ const configModule = require('../config/config');
  * @returns {Object} - config object. An error object is returned if this
  *  function is called with the error as the first argument.
  */
-function handleHeartbeatResponse(err, res) {
+function handleHeartbeatResponse(err, res, refocusInstanceObj=null) {
   debug('entered handleHeartbeatResponse');
   if (err) {
     logger.error('The handleHeartbeatResponse function was called with an ' +
       'error:', err);
     return err;
   }
 
+  // queue generation
+  // get queue
+  const bulkUpsertSampleQueue = queueUtils.getQueue('bulkUpsertSampleQueue');
+  if (bulkUpsertSampleQueue) {
+    if (res.collectorConfig) {
+      bulkUpsertSampleQueue._size = res.collectorConfig.maxSamplesPerBulkRequest;
+      bulkUpsertSampleQueue._flushTimeout =
+        res.collectorConfig.sampleUpsertQueueTime;
+    }
+  } else {
+    const config = configModule.getConfig();
+    queueUtils.createQueue('bulkUpsertSampleQueue',
+      config.collectorConfig.maxSamplesPerBulkRequest,
+      config.collectorConfig.sampleUpsertQueueTime,
+      false,
+      httpUtils.doBulkUpsert,
+      refocusInstanceObj
+    );
+  }
+
```

Comment:
What do you think about wrapping line 37 to 56 into a function and putting it in `src/heartbeat/utils.js
` ?  Also for line 47 to 54 when the heartbeat is received for the first time, config.collectorConfig is going to be null, you want to do this after `updateCollecorConfig` is done.

Reply:
This portion is unclear until we implement register or start command. Then we can fix this. What do you think? After that story we will have proper idea. @shriramshankar